src/shogun/mathematics/linalg/linsolver/DirectDenseLeastSquareSolverQR.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General turalPublic License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Written (W) 2015 Yingrui Chang
+ */
+
+#ifndef DIRECT_DENSE_LEAST_SQUARE_SOLVER_QR_H_
+#define DIRECT_DENSE_LEAST_SQUARE_SOLVER_QR_H_
+
+#include <shogun/lib/config.h>
+
+#ifdef HAVE_EIGEN3
+#include <shogun/mathematics/linalg/linsolver/LinearSolver.h>
+
+namespace shogun
+{
+
+/** @brief Class that provides a solve method for least square problem
+ * using QR decomposition
```

Comment:
Some math details would be good, as well as a wikipedia link.
Also, state the assumptions to the operator